### DISCUSSION TITLE
Security onion salt timeout error

### DISCUSSION THREAD

**Discussion by Tredmond7511** [Security-Onion-Solutions/securityonion#8582] [Unsupported Versions]
I'm trying to deploy forward sensors for a distributed deployment and all of them are failing during the installation script with a salt master timeout error. They can authenticate to the manager node but they all fail with the same error.

**Comment by dougburks** [comment]
Please make sure that sensors can connect to the manager on all ports listed here:
https://docs.securityonion.net/en/2.3/firewall.html#node-communication

**Reply by Tredmond7511** [reply]
Will do, I think I fixed the salt issue by adding the sensor ip in the saltstack minions configuration file. Now the install script fails because of yum repositories. I'm installing in an airgapped network. I tried disabling the repos, deleting them but the pre flight checks will still fail and the repos come right back. Any ideas?

**Reply by dougburks** [reply]
Are you using our latest ISO image for the manager and sensors?
https://blog.securityonion.net/2022/08/security-onion-23150-now-available.html

Are you using the `Airgap` option?
![Screen Shot 2022-08-25 at 6 37 19 AM](https://user-images.githubusercontent.com/1659467/186643700-ccb739cf-f85f-4792-bf6c-f9ff3293a61b.png)

What is the output of the following on your manager?
```
sudo grep airgap /opt/so/saltstack/local/pillar/global.sls 
```

For more information, please see:
https://docs.securityonion.net/en/2.3/airgap.html

**Reply by Tredmond7511** [reply]
I'm using .130 and my airgap under global.sls states true

**Reply by Tredmond7511** [reply]
I've gone through sosetup.log and see a couple errors under test.fail_without_changes for ssl state not allowed, firewall state not allowed, common state not allowed, nginx state not allowed, pcap state not allowed, suricata state not allowed, zeek state not allowed

**Reply by dougburks** [reply]
Seems like the sensor is having trouble pulling some information from the manager. Did you verify that the sensor can connect to the manager on all ports listed at https://docs.securityonion.net/en/2.3/firewall.html#node-communication?
